### PR TITLE
MINOR: [C++] Fix typos in key_map doc and comment

### PR DESCRIPTION
**Pull request by zanmato1984**

### CODE REVIEW

File: cpp/src/arrow/acero/doc/key_map.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ Every slot can either be **empty** or store data related to a single inserted ke
...
...o 8 inserted keys. When the first key is inserted into an empty block, it will occupy a slot with local id 0. The second inserted key will go into slot number 1 and so on. We use N highest bits of hash to get an index of a **start block**, when searching for a match or an empty slot to insert a previously not seen key when that is the case. If the start block contains any empty slots, then the search for either a match or place to insert a key will end at that block. We will call such a block an **open block**. A block that is not open is a full block. In the case of full block, the input key related search may continue in the next block modulo the number of blocks. If the key is not inserted into its start block, we will refer to it as an **overflow** entry, other entries being **non-overflow**. Overflow entries are slower to process, since they require visiting more than one block, so we want to keep their percentage low. This is done by choosing the right **load factor** (percentage of occupied slots in the hash table) at which the hash table gets resized and the number of blocks gets doubled. By tuning this value we can control the probability of encountering an overflow entry.
```
Comment (zanmato1984) [inline cpp/src/arrow/acero/doc/key_map.md:68]: module -> modulo

**Comment by zanmato1984** [comment]
cc @kou @pitrou

File: cpp/src/arrow/compute/key_map_internal.cc [hunk 2]
```diff
@@ -281,7 +281,7 @@ void SwissTable::early_filter_imp(const int num_keys, const uint32_t* hashes,
 // When we reach this limit, we need to break processing of any further rows and resize.
 //
 uint64_t SwissTable::num_groups_for_resize() const {
-  // Resize small hash tables when 50% full (up to 12KB).
+  // Resize small hash tables when 50% full (up to 32KB).
```
Comment (kou) [inline cpp/src/arrow/compute/key_map_internal.cc:284]: Sorry, how to compute `32KB`?

Comment (zanmato1984) [reply cpp/src/arrow/compute/key_map_internal.cc:284]: The `log_blocks_small = 9` means that we consider `1 << 9 = 512` blocks as "small". A block is a swiss-table-specific concept explained in https://github.com/apache/arrow/blob/main/cpp/src/arrow/acero/doc/key_map.md , which is 64-byte. So the small hash table ends up with `512 * 64B = 32KB`.

So the original `12KB` is obviously a typo.

Comment (kou) [reply cpp/src/arrow/compute/key_map_internal.cc:284]: Thanks!

**Review by kou — APPROVED** [review]
+1

**Comment by zanmato1984** [comment]
The CI failure is unrelated. I'm merging this.

Thank you @kou for looking!

**Pull request merged**